fulp_modules/fulp_world/public_restart_vote.dm

```diff
@@ -0,0 +1,67 @@
+#define CHOICE_RESTART "Call the shuttle"
+#define CHOICE_CONTINUE "Continue Playing"
+
+/datum/vote/public_shuttle_vote
+	name = "Shuttle call"
+	default_choices = list(
+		CHOICE_RESTART,
+		CHOICE_CONTINUE,
+	)
+	default_message = "Vote to call the shuttle and end the ongoing round."
+	var/last_held = 0
+
+/datum/vote/public_shuttle_vote/toggle_votable()
+	var/datum/config_entry/CE = /datum/config_entry/number/public_shuttle_vote
+	if(CONFIG_GET(number/public_shuttle_vote) == -1)
+		var/time_requirement = input(usr, "How many minutes should the round last before the vote can be called?", "Enabling restart vote", initial(CE.default)) as num
+		CONFIG_SET(number/public_shuttle_vote, time_requirement)
+	else
+		CONFIG_SET(number/public_shuttle_vote, -1)
+
+/datum/vote/public_shuttle_vote/is_config_enabled()
+	return  CONFIG_GET(number/public_shuttle_vote) != -1
+
+/datum/vote/public_shuttle_vote/create_vote()
+	last_held = world.time
+	. = ..()
+
+/datum/vote/public_shuttle_vote/get_vote_result()
+	. = ..()
+
+/datum/vote/public_shuttle_vote/can_be_initiated(forced)
+	var/time_remaining = CONFIG_GET(number/public_shuttle_vote) MINUTES - (world.time - SSticker.round_start_time)
+	var/can_evac = SSshuttle.canEvac()
+	var/cooldown = last_held ? (last_held + 20 MINUTES) - world.time : 0 // Always 0 if we haven't had a vote yet
+
+	. = ..()
+	if(. != VOTE_AVAILABLE)
+		return .
+	if(can_evac != TRUE)
+		return can_evac // Despite the name, it's either 'TRUE' OR a string describing why the shuttle can't be called.
+	if(forced)
+		return VOTE_AVAILABLE
+	if(time_remaining > 0)
+		return "Cannot be called until [DisplayTimeText(CONFIG_GET(number/public_shuttle_vote) MINUTES)] after the start of the round.\n\
+				([DisplayTimeText(time_remaining)] left)"
+	if(cooldown > 0)
+		return "A shuttle vote has already been called. Another will be available in [DisplayTimeText(cooldown)]."
+	return VOTE_AVAILABLE
+
+/datum/vote/public_shuttle_vote/finalize_vote(winning_option)
+	if(winning_option == CHOICE_CONTINUE)
+		return
+
+	if(winning_option == CHOICE_RESTART)
+		SSshuttle.emergency.request(reason = "\n\
+		\n\
+		S.M.I.L.E. telemetry shows morale has decayed beyond profitable limits. \
+		A mandatory crew rotation will take place, and janitorial staff have been dispatched to prepare the facilities for the next shift.\n\
```

Comment:
```suggestion
		A mandatory crew rotation will now take place. Crew remaining on site after the end of their shift may expect recovery in approximately six business weeks and are encouraged to apply for a Nanowage Overtime Plan Acclimated Yearly.\n\
```
I think that the current message might briefly mislead some players into thinking that a janitorial ERT is on the way. Honestly it would be pretty cool if the vote prompted admins with the option to send one on success, but for now it might just be better to change this section altogether.

Reply:
That was the intention, and I'm pending an answer on it. I quite like this alternative if it's turned down.

Reply:
Fair. If you do get that feature in then I would like to suggest the following adjustments:
```suggestion
		A mandatory crew rotation will now take place. Janitorial staff have been dispatched to prepare station facilities for the next shift.\n\
```

Reply:
I like the addition of 'now' and 'station'. I considered a full stop, but I don't quite like how the whole paragraph reads if it has three full stops in such quick succession.

Reply:
Well, in light of the fact it is *not* currently possible to call an ERT without making an entirely new proc for it, that functionality will be delayed until I make an upstream PR improving that. First suggestion is going in too.